src/app/wishlists/observers.ts

```diff
@@ -1,16 +1,20 @@
 import { set } from 'app/storage/idb-keyval';
-import { observeStore } from 'app/utils/redux';
+import { StoreObserver } from 'app/store/observerMiddleware';
+import { deepEqual } from 'fast-equals';
+import { WishListsState } from './reducer';
 
-export function saveWishListToIndexedDB() {
-  return observeStore(
-    (state) => state.wishLists,
-    (_prev, nextState) => {
-      if (nextState.loaded) {
+export function createWishlistObserver(): StoreObserver<WishListsState> {
+  return {
+    id: 'wish-list-observer',
+    equals: deepEqual,
```

Comment:
This can be the default Object.is. Our Redux state is only ever updated using immutable updates so reference equality is enough - if any sub-property changes the parents are all new references.